src/clients/SpokePoolClient.ts

```diff
@@ -799,10 +796,10 @@ export class SpokePoolClient extends BaseAbstractClient {
       // @note The type assertions here suppress errors that might arise due to incomplete types. For now, verify via
       // test that the types are complete. A broader change in strategy for safely unpacking events will be introduced.
       for (const event of fillEvents) {
-        const fill = this.isV3FillEvent(event)
-          ? { ...(spreadEventWithBlockNumber(event) as V3FillWithBlock), destinationChainId: this.chainId }
-          : { ...(spreadEventWithBlockNumber(event) as V2FillWithBlock) };
-
+        const fill = {
+          ...(spreadEventWithBlockNumber(event) as V3FillWithBlock),
```

Comment:
Out of scope question: I wonder if we can add an additional generic parameter on `spreadEventWithBlockNumber` so that we don't need to typecast as much

Reply:
Yeah; I think there's general consensus on the value of doing this :+1: